File: adt2sys/BlockWriteF.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void BlockWriteF (void *file, void *data, int32_t size, int32_t *bytes_written)
{
  DBG_ENTER ("BlockWriteF");

  Pascal_BlockWrite (file, data, size, bytes_written);
  if (Pascal_IOResult () != 0) *bytes_written = 0;

  DBG_LEAVE (); //EXIT //BlockWriteF
}

File: adt2sys/pas/BlockWriteF.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure BlockWriteF(var f: File; var data; size: Longint; var bytes_written: Longint);
begin
  _dbg_enter ({$I %FILE%}, 'BlockWriteF');

  {$i-}
  BlockWrite(f,data,size,bytes_written);
  {$i+}
  If (IOresult <> 0) then
    bytes_written := 0;

  _dbg_leave; //EXIT //BlockWriteF
end;
